src/lib/mongoBackend/mongoOntimeintervalOperations.cpp

```diff
@@ -80,22 +80,17 @@ HttpStatusCode mongoGetContextSubscriptionInfo
         csiP->entityIdVector.push_back(enP);
 
     }
-    std::vector<BSONElement> attrs = getField(sub, CSUB_ATTRS).Array();
-    for (unsigned int ix = 0; ix < attrs.size(); ++ix) {
-        csiP->attributeList.push_back(attrs[ix].String());
-    }
-
-    csiP->expiration = getField(sub, CSUB_EXPIRATION).numberLong();
 
-    csiP->url = getStringField(sub, CSUB_REFERENCE);
-    if (sub.hasElement(CSUB_LASTNOTIFICATION)) {
-        csiP->lastNotification = getIntField(sub, CSUB_LASTNOTIFICATION);
-    }
-    else {
-        csiP->lastNotification = -1;
+    std::vector<BSONElement> attrs = getField(sub, CSUB_ATTRS).Array();
+    for (unsigned int ix = 0; ix < attrs.size(); ++ix)
+    {
+      csiP->attributeList.push_back(attrs[ix].String());
     }
 
-    csiP->throttling = sub.hasField(CSUB_THROTTLING) ? getField(sub, CSUB_THROTTLING).numberLong() : -1;
+    csiP->url              = getStringField(sub, CSUB_REFERENCE);
+    csiP->expiration       = sub.hasField(CSUB_EXPIRATION)?       getIntOrLongFieldAsLong(sub, CSUB_EXPIRATION)       : -1;
+    csiP->lastNotification = sub.hasField(CSUB_LASTNOTIFICATION)? getIntOrLongFieldAsLong(sub, CSUB_LASTNOTIFICATION) : -1;
+    csiP->throttling       = sub.hasField(CSUB_THROTTLING)?       getField(sub, CSUB_THROTTLING).numberLong()         : -1;
```

Comment:
I think it could worth the effort... just in case we avoid some unexpected problem in the short future and it should be to hard to implemented.